Highway.cpp

```diff
@@ -13,22 +16,24 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     */
+    if( auto* car = dynamic_cast<Car*>(v) ) car->closeWindows();
+    if( auto* motorcycle = dynamic_cast<Motorcycle*>(v) ) motorcycle->lanesplitAndRace(100);
+    if( auto* truck = dynamic_cast<SemiTruck*>(v) ) truck->honkHorn();
```

Comment:
if it's a car, don't check if it's a truck or motorcycle.  you're still checking if it is, for all of the 3 choices.
use `else if`
- [ ] revise